core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql

```diff
@@ -59,6 +61,8 @@
       {% do to_drop.append(backup_relation) %}
   {% endif %}
 
+  {% do apply_grants(target_relation, grant_config, should_revoke=True) %}
```

Comment:
As we discussed yesterday, `should_revoke=True` will always work. For maximal efficiency, it should be conditional based on:
- Are we running this incremental model from scratch? (`existing_relation is none or full_refresh_mode`)
- Is this database one that carries over grants when fully replacing an object? (adapter-specific method to tell us `True` or `False`)